src/re_frame/loggers.cljc

```diff
@@ -0,0 +1,36 @@
+(ns re-frame.loggers
+  (:require
+    [clojure.set :refer [difference]]))
+
+;; "loggers" holds the current set of logging functions.
+;; By default, re-frame uses the functions provided by js/console.
+;; Use set-loggers! to change these defaults.
+
+(def ^:private loggers
+  (atom {:log       #?(:cljs (js/console.log.bind   js/console)
+                       :clj  println)
+         :warn      #?(:cljs (js/console.warn.bind  js/console)
+                       :clj  (partial println "WARN: "))
```

Comment:
I think it's probably better to use tools.logging so that you can control logging levels. It could be used either for certain tests that you know will cause an error to be logged that you don't want to be emitted, or just controlling the overall logging level at different times. Otherwise, you will get println's in your tests that you can't turn off.